src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2460,6 +2469,61 @@ PlanRouterQuery(Query *originalQuery,
 }
 
 
+/*
+ * AllShardsColocated returns true if all the shards in the given relationShardList
+ * have colocated tables and are on the same shard index.
+ */
+static bool
+AllShardsColocated(List *relationShardList)
+{
+	RelationShard *relationShard = NULL;
+	int shardIndex = -1;
+	int colocationId = -1;
+	foreach_ptr(relationShard, relationShardList)
+	{
+		Oid relationId = relationShard->relationId;
+		uint64 shardId = relationShard->shardId;
+		if (shardId == INVALID_SHARD_ID)
+		{
+			/* intermediate results are always colocated */
+			continue;
+		}
+
+		if (!(IsCitusTableType(relationId, HASH_DISTRIBUTED) ||
+			  IsCitusTableType(relationId, SINGLE_SHARD_DISTRIBUTED)))
```

Comment:
seems trivial to add range-distributed here

Reply:
Not really, because ShardIndex function cannot hanle range distributed tables.

I initially thought that is probably a limitation of `ShardIndex` function, but it looks like we never support colocation of rangeTable withing the code? For example: https://github.com/citusdata/citus/blob/c2f46f0f3f3a74c7a6ce4b37b2d2797643db010a/src/backend/distributed/utils/colocation_utils.c#L411-L460 

Or here: https://github.com/citusdata/citus/blob/c2f46f0f3f3a74c7a6ce4b37b2d2797643db010a/src/backend/distributed/utils/colocation_utils.c#L1044-L1057


Overall, I think I don't care much about range partitioned tables :D  So, I'm fine for what you suggest. But, I'd be a little cautious to change how code treats range partitioned tables in the code.